varya/assets/js/customizer-validate-wcag-color-contrast.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * @author Per Soderlind
+ * @link https://github.com/soderlind/2016-customizer-demo
+ * global wp, _validateWCAGColorContrastExports
+ * exported validateWCAGColorContrast
+**/
+( function( $, api, exports ) {
+	var self = {
+		validate_color_contrast: []
+	};
+	if ( exports ) {
+		$.extend( self, exports );
+	}
+
+	/**
+	 * Add contrast validation to a control if it is entitled (is a valid color control).
+	 *
+	 * @param {wp.customize.Control} setting                   - Control.
+	 * @param {wp.customize.Value}   setting.validationMessage - Validation message.
+	 * @return {boolean} Whether validation was added.
+	 */
+	self.addWCAGColorContrastValidation = function( setting ) {
+		var initialValidate;
+		if ( ! self.isColorControl( setting ) ) {
+			return false;
+		}
+
+		initialValidate = setting.validate;
+
+		/**
+		 * Wrap the setting's validate() method to do validation on the value to be sent to the server.
+		 *
+		 * @param {mixed} value   - New value being assigned to the setting.
+		 * @returns {*}
+		 */
+		setting.validate = function (value){
+			var failsWCAG = false;
+			var code = 'contrast_warning';
+
+			_.each ( self.validate_color_contrast[setting.id], function( color_to_compare_id ){
+				var color_to_compare = $( '#customize-control-' + color_to_compare_id + ' .wp-color-result')[0].style.backgroundColor.match(/\d+/g);
+				var contrast	= self.rgb( [ parseInt(color_to_compare[0]), parseInt(color_to_compare[1]), parseInt(color_to_compare[2]) ], self.hexRgb( value ) );
+				var score		= self.score( contrast );
+				if ( score === 'Fail' ){
+					failsWCAG = true;
+				} 
+			});
+
+			if ( failsWCAG ){
+				var validationWarning = new api.Notification( code, { message: 'This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.', type: 'warning' } );
```

Comment:
We may need to do some fancy PHP/JS translation escaping here so that this text can be translated more easily. Its kind of a annoying to do, but I’m sure it’s something the TRT team would catch. I’ll send you some links that show how we do it.

Also wondering if we should change `Try using a brighter background color and/or a darker text color.` to this: `Try using a brighter background color and/or a darker **foreground** color.` since the text color is called foreground in the UI.